data/random-teams.ts

```diff
@@ -676,7 +676,7 @@ export class RandomTeams {
 				let moveType = move.type;
 				if (['judgment', 'revelationdance'].includes(moveid)) moveType = types[0];
 				if (!this.noStab.includes(moveid) && (move.basePower > 30 || move.multihit || move.basePowerCallback)) {
-					if (teraType === moveType) {
+					if (teraType === moveType && species.baseSpecies !== 'Dudunsparce') {
```

Comment:
```suggestion
					if (teraType === moveType) {
```

Dudunsparce check would be better placed in line 672